lib/memory-store.js

```diff
@@ -9,7 +10,11 @@ function MemoryStore(windowMs) {
           hits[key] = 1;
       }
 
-      cb(null, hits[key]);
+      if (!timeReset[key]) {
+          timeReset[key] = parseInt((new Date().getTime() / 1000).toFixed(0), 10) + (windowMs / 1000);
```

Comment:
This isn't quite correct. For the MemoryStore, all IPs are reset at once, so the offset should be from the last time `resetAll()` was called, not from the time of the user's first request. 

Because of this, it is unnecessary to track or even calculate the time remaining on a per-IP basis. Just record a single `var nextReset = Math.ceil((Date.now()+windowMs)/1000)`, and then update it each time `resetAll()` is called. When a new hit comes in, the reset time is always `nextReset` no matter when we first saw that IP.

(`Date.now()` is the same as `(new Date()).getTime()` but shorter and more efficient ;)